tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -647,6 +647,20 @@ def test_get_best_booster(self) -> None:
         with pytest.raises(ValueError):
             tuner2.get_best_booster()
 
+    @pytest.mark.parametrize("mock_exists, expected", [(False, True), (True, False)])
+    def test_model_dir(self, mock_exists: bool, expected: bool) -> None:
+        params: Dict = {"verbose": -1}
+        dataset = lgb.Dataset(np.zeros((10, 10)))
+
+        study = optuna.create_study()
+
+        with mock.patch("optuna.integration._lightgbm_tuner.optimize.os.mkdir") as m:
+            with mock.patch("os.path.exists", return_value=mock_exists):
+                LightGBMTuner(
+                    params, dataset, valid_sets=dataset, study=study, model_dir="./booster"
+                )
```

Comment:
I think we can remove `study` since it will be created inside `LightGBMTuner`.
```suggestion

        with mock.patch("optuna.integration._lightgbm_tuner.optimize.os.mkdir") as m:
            with mock.patch("os.path.exists", return_value=mock_exists):
                LightGBMTuner(
                    params, dataset, valid_sets=dataset, model_dir="./booster"
                )
```

Reply:
Thank you for pointing out! I have removed `study` as you mentioned.